### DISCUSSION TITLE
Why not export your caching model into a separate module?

### DISCUSSION THREAD

**Discussion by AmrSaber** [vercel/swr#930] [Ideas]
Hello, guys.

I am very pleased with the simplicity of using `SWR` and how you do great work in managing the cache. Recently, I used SWR to manage persistent state by defining a custom fetcher that fetches data from local storage. And things are very simple and are working great for me.

I have been thinking though: Your caching model and simple, intuitive interface are very easy and make the project and the code very simple, so why not publish a version of SWR to be used to manage and publish local state (to be shared across different screens easily without persistent). It would be a great way to manage a simple local state in the place of some more complex state management libraries like Redux, pull state and so when all you need is simply sharing some state.

**Comment by koba04** [comment]
There is a discussion about this at https://github.com/vercel/swr/pull/918 and https://github.com/vercel/swr/issues/738.

**Comment by AmrSaber** [comment]
We just published a package (https://github.com/zydalabs/swr-internal-state) that uses `SWR` and builds on top of it 2 hooks for handling internal state. One of them for the general global state (as in my question) and the other for handling the persistent state (the example I mentioned in the question).

The package is manually tested and the base implementation is done, documentation is a work in progress, and the automated tests are in the TODO. Any contribution is totally welcome and actually much appreciated.
